src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -69,27 +83,43 @@
     }
 
     public static @Nonnull GitHub connect(@CheckForNull String apiUri, @CheckForNull StandardCredentials credentials) throws IOException {
-        if (Util.fixEmptyAndTrim(apiUri) == null) {
-            if (credentials == null) {
-                return new GitHubBuilder().withRateLimitHandler(CUSTOMIZED).build();
-            } else if (credentials instanceof StandardUsernamePasswordCredentials) {
-                StandardUsernamePasswordCredentials c = (StandardUsernamePasswordCredentials) credentials;
-                return GitHub.connectUsingPassword(c.getUsername(), c.getPassword().getPlainText());
-            } else {
-                // TODO OAuth support
-                throw new IOException("Unsupported credential type: " + credentials.getClass().getName());
-            }
+        GitHubServerConfig config = new GitHubServerConfig(credentials!=null ? credentials.getId() : null);
+        String apiUrl = Util.fixEmptyAndTrim(apiUri);
+        if (apiUrl !=null) {
+            config.setCustomApiUrl(true);
+            config.setApiUrl(apiUrl);
+        }
+
+// Can't do this until github plugin support username/password
```

Comment:
Github-plugin provides ability to convert login+pwd to token easily. So it any other can point to this ability if anybody wants to use login and password